hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java

```diff
@@ -658,6 +658,19 @@ public String getUsername() {
     return userName;
   }
 
+  /**
+   * Get the root directory of Trash for a path in OFS.
+   * Returns /<volumename>/<bucketname>/.Trash/<username>
+   * Caller appends either Current or checkpoint timestamp for trash destination
+   * @param path the trash root of the path to be determined.
+   * @return trash root
+   */
+  @Override
```

Comment:
You also need to Override getTrashRoots() which returns all trash locations under the root across multiple buckets. This should be very similar to how Trash under encryption zone is handled in DistributedFileSystem#getTrashRoots().

Reply:
Good catch. It also seems [`Emptier`](https://github.com/apache/hadoop/blob/f86de6f76a3079c2655df9b242fc968edfb17b9d/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/TrashPolicyDefault.java#L288) in `TrashPolicyDefault` utilizes `getTrashRoots()`, which @umamaheswararao believes might make trash clean up easier in Ozone.

Reply:
Created HDDS-3705 to track `getTrashRoots` impl.